modules/core/include/opencv2/core/hal/intrin_avx.hpp

```diff
@@ -1481,6 +1481,22 @@ inline v_int8x32 v_absdiffs(const v_int8x32& a, const v_int8x32& b)
     v_int8x32 m = a < b;
     return (d ^ m) - m;
 }
+
+static inline v_uint8x32 v_absdiffs(const v_uint8x32& a, const v_uint8x32& b)
+{
+    return v_max(a, b) - v_min(a, b);
+}
+
+static inline v_uint16x16 v_absdiffs(const v_uint16x16& a, const v_uint16x16& b)
+{
+    return v_max(a, b) - v_min(a, b);
+}
+
+static inline v_float32x8 v_absdiffs(const v_float32x8& a, const v_float32x8& b)
+{
+    return v_max(a, b) - v_min(a, b);
+}
+
 inline v_int16x16 v_absdiffs(const v_int16x16& a, const v_int16x16& b)
 { return v_max(a, b) - v_min(a, b); }
```

Comment:
why not use a template for this ?

Reply:
I can't apply templates because there are v_absdiffs() function with different implementation. For example:

```
inline v_int8x64 v_absdiffs(const v_int8x64& a, const v_int8x64& b)
{
    v_int8x64 d = a - b;
    v_int8x64 m = a < b;
    return (d ^ m) - m;
}
```

Reply:
IMO it doesn't make sense to introduce new intrinsics here. For unsigned types absolute difference of values is always inside the representable range. So the result of `v_absdiffs` will be always equal to `v_absdiff`. For floating values saturation is a thing to discuss, however this implementation won't do it anyway.

Reply:
Ok. Done.